TripalCultivate/TripalCultivate-Phenotypes
# Update docker to import content types + run tripal job.

**Issue #55**

## Motivation

When PR https://github.com/tripal/tripal/pull/1696 was merged, it changes the purpose of prepare and how content types were created. This means, our current dockerfile is not creating any content types at all... 🤪 This hasn't been a problem yet but it is good to solve anyway.

## What does this PR do?
*Please describe each things this PR does. For example, a PR may 1) solve a specific bug, 2) create an auomated test to ensure it doesn't return.*

1. Adds drush commands to the dockerfile to import the General + Germplasm content type collections in core.
2. Adds drush command to dockerfile to run tripal jobs. This will run the job added when this module is enabled.

## Testing

### Manual Testing
*Describe in detail how someone should manually test this functionality.*
*Make sure to include whether they need to build a docker from scratch, create any records, etc.*

1. Clone this repo and switch to this branch:
```
git clone https://github.com/TripalCultivate/TripalCultivate-Phenotypes trpcult-pheno-g0.55-importContentTypes
cd trpcult-pheno-g0.55-importContentTypes
git checkout g0.55-importContentTypes
```
2. Create a new docker image: 
```
docker build --tag=trpcultivate-pheno:G0.55 ./
```
3. Create a new container using that fresh image:
```
docker run --publish=80:80 -tid --name=phenoG0.55 --volume=/Users/las166/Docker/trpcult-pheno-g0.55-importContentTypes:/var/www/drupal/web/modules/contrib/TripalCultivate-Phenotypes trpcultivate-pheno:G0.55
docker exec phenoG0.55 service postgresql restart
```
4. Navigate to localhost and login to the site.
5. Go to Admin > Tripal > Page Structure and confirm that there are the following content types.
![TripalContentTypeListing](https://github.com/TripalCultivate/TripalCultivate-Phenotypes/assets/1566301/42d5550d-190c-402e-b5e8-4a7ab63c13d7)
6. Go to Admin > Tripal > Extensions > Tripal Cultivate Phenotypes Configuration > Ontology Terms and confirm that it doesn't give you a warning to run the tripal job. Rather it will warn you to create an organism ;-) 
7. Create an organism and then go back to the ontology configuration page and ensure it allows you to configure!
